### PR TITLE
updates sample yaml files with right apiVersion and updates CRDs to v1 from v1 beta

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
* This PR fixes the apiVersion in sample yaml files and also fixes a few fields values.
* It also fixes the warning caused due to CRDs using v1beta which will be discontinued from K8 v1.22+ by updating it to v1
* It fixes the conformity of the YAML files in crd folder according to v1 CRDs

### CODE REVIEW

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
Hi @mritunjaysharma394 - the YAML CRD files are generated from the Golang structures in https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/api/v1alpha1/policyreport_types.go. So, we do not need to update them manually but update the structs and then regenerate the CRD YAMLs.

**Comment by mritunjaysharma394** [comment]
> Hi @mritunjaysharma394 - the YAML CRD files are generated from the Golang structures in https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/api/v1alpha1/policyreport_types.go. So, we do not need to update them manually but update the structs and then regenerate the CRD YAMLs.

Hi @JimBugwadia , as discussed and suggested, I have updated the controller-gen from its existing version in Makefile to the latest 0.5.0. 
As a result of this, we have our CRDs apiVersion updated to v1 from v1beta automatically. 
Also, I have fixed the conformity issue in the sample YAML files. 
The installation now runs without warnings and errors. 
Screenshot from 2021-03-16 13-35-50

Thank you so much for your feedback!

File: policy-report/samples/sample-cis-k8s.yaml [hunk 1]
```diff
@@ -12,28 +12,15 @@ summary:
...
   fail: 2
   warn: 0
-  info: 0
   error: 0
   skip: 0
 results:
   - policy: api-server:anonymous-auth
     message: ensure that --anonymous-auth argument is set to false
-    status: warn
     scored: true
```
Comment (JimBugwadia) [inline policy-report/samples/sample-cis-k8s.yaml:20]: Why did we remove the `status`? I believe that is a required field right?

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
please see question on status field in the samples

File: policy-report/samples/sample-cis-k8s.yaml (hunk 1 above)
Comment (mritunjaysharma394) [reply policy-report/samples/sample-cis-k8s.yaml:20]: I guess if we keep the `status` as a field in the results section, we are getting the following error:
```error: error validating "samples/sample-cis-k8s.yaml": error validating data: ValidationError(PolicyReport.results[0]): unknown field "status" in io.wgpolicyk8s.v1alpha1.PolicyReport.results; if you choose to ignore these errors, turn validation off with --validate=false``` 

Besides that, I was not able to find `status` in https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/api/v1alpha1/policyreport_types.go, so please tell what can we figure out from these and what should we do?

Comment (mritunjaysharma394) [reply policy-report/samples/sample-cis-k8s.yaml:21]: Thanks @JimBugwadia , changed `status` to `result` and now the above errors have gone. Thanks a lot!

File: policy-report/samples/sample-rhacm-policy.yaml [hunk 2]
```diff
@@ -16,16 +15,10 @@ summary:
...
   message: must not have imagemanifestvulns
-  status: Fail
+  result: fail
   scored: false
   resources:
     - apiVersion: secscan.quay.redhat.com/v1alpha1
       kind: ImageManifestVuln
       name: sha256.8d104847fc2371a983f7cb01c7c0a3ab35b7381d6bf7ce355d9b32a08c0031f0
       namespace: openshift-cluster-version
-  data:
```
Comment (JimBugwadia) [inline policy-report/samples/sample-rhacm-policy.yaml:26]: @mritunjaysharma394 - please add back `data` as it is a supported field.

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
One more change requested, before we can merge.

File: policy-report/samples/sample-rhacm-policy.yaml (hunk 2 above)
Comment (mritunjaysharma394) [reply policy-report/samples/sample-rhacm-policy.yaml:26]: While adding `data`, I was getting this error: 

```
error: error validating "samples/sample-rhacm-policy.yaml": error validating data: ValidationError(PolicyReport.results[0]): unknown field "data" in io.wgpolicyk8s.v1alpha1.PolicyReport.results; if you choose to ignore these errors, turn validation off with --validate=false

```

On checking this again (https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/api/v1alpha1/policyreport_types.go), I realised that `data`  might have been renamed as `properties`. Should I add `data` as `properties`?

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**